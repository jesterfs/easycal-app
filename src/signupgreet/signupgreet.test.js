import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import SignupGreet from './signupgreet'
import renderer from 'react-test-renderer';

describe('SignupGreet component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <SignupGreet />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const div = document.createElement('div')
    const tree = renderer
      .create(<BrowserRouter>
        <SignupGreet />
      </BrowserRouter>,
      div)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

});