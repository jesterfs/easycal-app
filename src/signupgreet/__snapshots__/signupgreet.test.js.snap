// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupGreet component renders the UI as expected 1`] = `
<div
  className="SignUpGreet, greetgroup"
>
  <div
    className="item"
  >
    <h2>
      Create Your EasyCal Account
    </h2>
  </div>
  <div
    className="item"
  >
    <form
      className="signup-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="signupName"
        >
          Full Name
        </label>
        <input
          id="signupName"
          name="signupName"
          placeholder="Full Name"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="signupEmail"
        >
          Email
        </label>
        <input
          id="signupEmail"
          name="signupEmail"
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="signupPassword"
        >
          Password
        </label>
        <input
          id="signupPassword"
          name="signupPassword"
          type="password"
        />
      </div>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;
